pkg/instance/instance.go

```diff
@@ -406,122 +413,187 @@ func (s *Server) spdkInstanceLog(req *rpc.InstanceLogRequest, srv rpc.InstanceSe
 	return grpcstatus.Error(grpccodes.Unimplemented, "spdk instance log is not supported")
 }
 
+func (s *Server) handleNotify(ctx context.Context, notifyChan chan struct{}, srv rpc.InstanceService_InstanceWatchServer) error {
+	logrus.Info("Start handling notify")
+
+	for {
+		select {
+		case <-ctx.Done():
+			logrus.Info("Stopped handling notify")
+			return ctx.Err()
+		case <-notifyChan:
+			if err := srv.Send(&empty.Empty{}); err != nil {
+				return errors.Wrap(err, "failed to send instance response")
+			}
+		}
+	}
+}
+
 func (s *Server) InstanceWatch(req *empty.Empty, srv rpc.InstanceService_InstanceWatchServer) error {
 	logrus.Info("Start watching instances")
 
+	// Create a client for watching processes
+	pmClient, err := client.NewProcessManagerClient("tcp://"+s.processManagerServiceAddress, nil)
+	if err != nil {
+		return grpcstatus.Error(grpccodes.Internal, errors.Wrapf(err, "failed to create ProcessManagerClient").Error())
+	}
+	defer pmClient.Close()
+
+	var spdkClient *spdkclient.SPDKClient
+	if s.spdkEnabled {
+		// Create a client for watching SPDK engines and replicas
+		spdkClient, err = spdkclient.NewSPDKClient(s.spdkServiceAddress)
+		if err != nil {
+			return grpcstatus.Error(grpccodes.Internal, errors.Wrapf(err, "failed to create SPDK client").Error())
+		}
+		defer spdkClient.Close()
+	}
+
+	notifyChan := make(chan struct{})
```

Comment:
NIT: Do we need to make this a buffered one?

Reply:
Yeah, buffered channel sounds good. Use `make(chan struct{}, 1024)` instead.